### DISCUSSION TITLE
google cloud storage at .env

### DISCUSSION THREAD

**Discussion by iqraaaswad** [directus/directus#8981] [Archive]
what's exactly STORAGE_<LOCATION>_KEY_FILENAME at google cloud storage ? is it secret key or access key ? could u give me full example about gcs at .env file ?

**Comment by rijkvanzanten** [comment]
When using GCS, Google might provide you with a `key.json` based on your configuration. This env var tells the underlying system where to find this key.json file 👍🏻

**Reply by iqraaaswad** [reply]
okay.. i got it.. thanks @rijkvanzanten

**Reply by AxeemHaider** [reply]
what if you already deploying directus on google cloud run?

**Comment by dunods** [comment]
i got an error saying that 

 A bucket name is needed to use Cloud Storage.

how do i resolve?

**Reply by rijkvanzanten** [reply]
Did you configure both environment variables? 

https://docs.directus.io/configuration/config-options/#google-cloud-storage-gcs

**Reply by AxeemHaider** [reply]
this URL is not working

**Reply by rijkvanzanten** [reply]
@AxeemHaider https://docs.directus.io/self-hosted/config-options.html#google-cloud-storage-gcs
